digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/csvDataImporter/TrafficSignCsvImporter.scala

```diff
@@ -89,20 +108,67 @@ class TrafficSignCsvImporter(roadLinkServiceImpl: RoadLinkService, eventBusImpl:
     }
   }
 
+  override def verifyData(parsedRow: ParsedProperties, user: User): ParsedCsv = {
+    val optLon = getPropertyValueOption(parsedRow, "lon").asInstanceOf[Option[BigDecimal]]
+    val optLat = getPropertyValueOption(parsedRow, "lat").asInstanceOf[Option[BigDecimal]]
+
+    val optTrafficSignType = getPropertyValueOption(parsedRow, "trafficSignType").asInstanceOf[Option[Int]]
+    val optStartDate = getPropertyValueOption(parsedRow, "startDate").asInstanceOf[Option[String]]
+    val optEndDate = getPropertyValueOption(parsedRow, "endDate").asInstanceOf[Option[String]]
+
+    val isValidDate = optTrafficSignType match {
+      case Some(value) =>
+        val isRoadworks = TrafficSignType.applyTRValue(value) == RoadWorks
+          (optStartDate, optEndDate) match {
+            case (Some(startDate), Some(endDate)) =>
+                  try {
+                    val startDateFormat = dateFormatter.parseDateTime(startDate)
+                    val endDateFormat = dateFormatter.parseDateTime(endDate)
+
+                    endDateFormat.isAfter(startDateFormat) || endDateFormat.isEqual(startDateFormat)
+                  } catch {
+                    case _: Throwable => !isRoadworks
+                  }
+            case _ => !isRoadworks
+          }
+      case _ => true
+    }
+
+    if(isValidDate) {
```

Comment:
If the dates are invalid, the process needs to continue validate the closestlink